src/jenkins_job_insight/analyzer.py

```
@@ -235,11 +250,35 @@ async def call_ai_cli(
+
+    if config.supports_stdin:
+        # Pass prompt via stdin (no ARG_MAX limit)
+        stdin_input = prompt
+    else:
+        # Append prompt as positional arg with safety truncation
+        prompt_encoded = prompt.encode("utf-8")
+        prompt_bytes = len(prompt_encoded)
+        if prompt_bytes > MAX_CLI_ARG_BYTES:
+            logger.warning(
+                f"Prompt size ({prompt_bytes} bytes) exceeds safe CLI argument limit "
+                f"({MAX_CLI_ARG_BYTES} bytes). Truncating for {provider_info}."
```

Ruff (0.14.14)

[warning] Logging statement uses f-string

(G004)